tools/nnyeah/nnyeah/Reworker.cs

```diff
@@ -0,0 +1,481 @@
+using System;
+using System.IO;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Rocks;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace nnyeah {
+	public class Reworker {
+		Stream stm;
+		ModuleDefinition module;
+		TypeDefinition embeddedAttributeTypeDef;
+		TypeDefinition nativeIntegerAttributeTypeDef;
+		TypeReference nativeIntegerAttributeTypeRef;
+		TypeReference compilerGeneratedAttributeTypeRef;
+		TypeReference embeddedAttributeTypeRef;
+		TypeReference nintTypeReference;
+		TypeReference nuintTypeReference;
+
+		Dictionary<string, Transformation> methodSubs;
+		Dictionary<string, Transformation> fieldSubs;
+
+		public Reworker (Stream stm)
+		{
+			this.stm = stm;
+		}
+
+		public void Load ()
+		{
+			if (module != null)
+				return;
+			module = ModuleDefinition.ReadModule (stm);
+		}
+
+		public bool NeedsReworking ()
+		{
+			// simple prediate for seeing if there any references
+			// to types we need to care about.
+			// IntPtr is for future handling of types that
+			// descend from NObject and need to have the constructor
+			// changed to NHandle
+			return module.GetTypeReferences ().Any (
+				tr =>
+					tr.FullName == "System.nint" ||
+					tr.FullName == "System.nuint" ||
+					tr.FullName == "System.nfloat" ||
+					tr.FullName == "System.IntPtr"
+				);
+		}
+
+		void CheckModule ()
+		{
+			if (module == null)
+				throw new Exception ("Module is not loaded. Call Load first.");
+		}
+
+		public void Rework (Stream stm)
+		{
+			CheckModule ();
+
+			// get the types that we need to refer to later
+			FetchCompilerGeneratedAttributeTypeRef ();
+			AddEmbeddedAttributeIfNeeded ();
+			AddNativeIntegerAttributeIfNeeded ();
+			module.TryGetTypeReference ("System.nint", out nintTypeReference);
+			module.TryGetTypeReference ("System.nuint", out nuintTypeReference);
```

Comment:
You are using a TryGetTypeReference which will return a boolean, meaning that the our parameters can be null, yet you do not check either the boolean call of the nullability of the output param. Should we check that? What happens if one of them is null?

Reply:
We don't need to check it. If the type returns null, it means that it will never match anything (ie, the type is not referenced in the module). It's a non-issue.